components/root/root.jsx

```diff
@@ -146,8 +147,74 @@ export default class Root extends React.Component {
         };
     }
 
+    getSidebarCategories = () => {
+        const sidebarCategories = [
+            document.getElementById('unreadsChannelList'),
+            document.getElementById('favoriteChannelList'),
+            document.getElementById('publicChannelList'),
+            document.getElementById('privateChannelList'),
+            document.getElementById('directChannelList'),
+        ].filter((element) => Boolean(element));
+
+        return sidebarCategories;
+    }
+
     handleAccessibilityKeys = (e) => {
-        if (isKeyPressed(e, Constants.KeyCodes.TAB)) {
+        if (isKeyPressed(e, Constants.KeyCodes.DOWN)) {
+            const lhsList = document.getElementById('lhsList');
+            const sidebarCategories = this.getSidebarCategories();
+            if (lhsList.contains(e.target)) {
+                const containsElement = (element, i, elementsList) => {
+                    if (element.contains(e.target)) {
+                        this.currentSidebarFocus = i + 1;
+                        return true;
+                    } else if (element.contains(e.target) && i === elementsList.length - 1) {
+                        this.currentSidebarFocus++;
+                        return false;
+                    }
+                    return false;
+                };
+
+                if (!sidebarCategories.some(containsElement)) {
+                    this.currentSidebarFocus = 0;
+                }
+
+                if (this.currentSidebarFocus === sidebarCategories.length) {
+                    this.currentSidebarFocus = 0;
+                }
+
+                const activeElement = sidebarCategories[this.currentSidebarFocus];
+                this.addFocusClass(activeElement);
+                this.currentSidebarFocus++;
+            }
+        } else if (isKeyPressed(e, Constants.KeyCodes.UP)) {
+            const lhsList = document.getElementById('lhsList');
+            const sidebarCategories = this.getSidebarCategories();
+            if (lhsList.contains(e.target)) {
```

Comment:
Could we check the opposite and return here if it passes? eg.

```javscript
if (!lhsList.contains(e.target)) {
  return
}
// do stuff
```

Makes it a lot cleaner and easier to read